modules/todo/components/TodoForm.tsx

```
@@ -0,0 +1,54 @@
+import React, { useState } from "react";
+import { FormItem } from "../../../shared/components/ui/form";
+import { Button } from "../../../shared/components/ui/button";
+import { Input } from "../../../shared/components/ui/input";
+import { useAppDispatch } from "@/shared/redux/hooks";
+import { TodoSchema } from "../models";
+import { addTodo } from "@/shared/redux/reducers/todoSlice";
+
+const TodoForm: React.FC = () => {
+  const [text, setText] = useState("");
+  const dispatch = useAppDispatch();
+
+  const handleAdd = (event: React.FormEvent) => {
+    event.preventDefault();
+    const result = TodoSchema.safeParse({
+      id: Date.now(),
+      task: text,
+      isCompleted: false,
+      priority: "No",
+      deadline: null,
+    });
+    if (result.success) {
+      dispatch(addTodo(result.data));
+      console.log(result.data);
+      setText("");
+    } else {
+      console.error(result.error);
```

same here....why errors are not handled properly??